src/internal/utils/handle-key-down.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { KeyCode } from '../keycode';
+
+export default function handleKeyDown({
```

Comment:
Does the writing direction affect all possible usages of this abstraction? For example, there can be cases where we just want to move the focus one element to the left or to the right, regardless of the writing direction, so handlers like `onLeft` and `onRight` might make sense too. Maybe it makes sense to start only with the handlers that we need, and keep adding the ones that we see necessary. I wouldn't mind mixing logical and directional handlers if we see that both have their sensible use cases.

Reply:
Can you elaborate on circumstances where we would want to go left in the UI for a component in both LTR and RTL writing modes?

Reply:
Actually if we manage logical handlers, not sure it is a "keyboard handler" anymore, but some other kind of abstraction. For example, `handleDirection` and it could be built on top of this one?